src/lib.rs

```diff
@@ -198,3 +198,36 @@ impl TryFrom<&sev::Chain> for Generation {
         })
     }
 }
+
+/// A description of the SEV-SNP platform's build information.
+#[repr(C)]
+#[derive(Copy, Clone, Debug, Default, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize)]
+pub struct SnpBuild {
+    /// The version information.
+    pub version: Version,
+
+    /// The build ID.
+    pub build: u32,
+}
+
+impl std::fmt::Display for SnpBuild {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        write!(f, "{}.{}", self.version, self.build)
+    }
+}
+
+impl codicon::Decoder<()> for SnpBuild {
+    type Error = std::io::Error;
+
+    fn decode(mut reader: impl Read, _: ()) -> std::io::Result<Self> {
+        reader.load()
+    }
+}
+
+impl codicon::Encoder<()> for SnpBuild {
+    type Error = std::io::Error;
+
+    fn encode(&self, mut writer: impl Write, _: ()) -> std::io::Result<()> {
+        writer.save(self)
+    }
+}
```

Comment:
I originally seen codicon traits implemented for the original `Build` and wanted `SnpBuild` to mirror the `Build` functionality. I'll remove them.

Reply:
iirc those were implemented to serialize/deserialize data to persistent storage (file) or over the wire since they're used in remote attestation messages.